src/main/scala/strawman/collection/mutable/Seq.scala

```diff
@@ -0,0 +1,76 @@
+package strawman
+package collection
+package mutable
+
+trait Seq[A] extends strawman.collection.Seq[A] with Growable[A] {
+  def update(idx: Int, elem: A): Unit
+  def insert(idx: Int, elem: A): Unit
+  def insertAll(idx: Int, elems: IterableOnce[A]): Unit
+  def remove(idx: Int): Option[A]
+  def remove(from: Int, n: Int): Unit
+  def mapInPlace(f: A => A): this.type
+  def flatMapInPlace(f: A => IterableOnce[A]): this.type
+  def filterInPlace(p: A => Boolean): this.type
+  def patchInPlace(from: Int, patch: collection.Seq[Int], replaced: Int): this.type
+
+  // +=, ++=, clear inherited from Growable
+  def +=:(elem: A): this.type = { insert(0, elem); this }
+  def +=:(elem1: A, elem2: A, elems: A*): this.type = elem1 +=: elem2 +=: elems.toStrawman ++=: this
+  def ++=:(elems: IterableOnce[A]): this.type = { insertAll(0, elems); this }
+
+  def dropInPlace(n: Int): this.type = { remove(0, n); this }
+  def dropRightInPlace(n: Int): this.type = { remove(length - n, n); this }
+  def takeInPlace(n: Int): this.type = { remove(n, length); this }
+  def takeRightInPlace(n: Int): this.type = { remove(0, length - n); this }
+  def sliceInPlace(start: Int, end: Int): this.type = takeInPlace(end).dropInPlace(start)
```

Comment:
`UnrolledBuffer` is another one that we have, and a `SortedSeq` backed by a tree would be another.  It might fit better in `IndexedOptimizedSeq`, since there presumably the exceptions would be vanishingly rare (and most everything else would be an `IndexedOptimizedSeq` anyway).